### PR TITLE
Fix user flairs and new comment indicators lost in rebase

### PR DESCRIPTION
**Pull request by zanderb98**
Fix flairs and new comment indicators that were lost in rebase. Streamable videos are still bonked but too lazy to fix now

### CODE REVIEW

File: winston/components/Badge.swift [hunk 1]
```diff
@@ -63,7 +83,7 @@ struct BadgeView: View, Equatable {
         Text(author).font(.system(size: theme.authorText.size, weight: theme.authorText.weight.t)).foregroundColor(author == "[deleted]" ? .red : usernameColor ?? theme.authorText.color.cs(cs).color())
           .onTapGesture(perform: openUser)
 
-                    if unseen {
+          if unseen {
             ZStack {
               Circle()
                 .fill(commentTheme?.unseenDot.cs(cs).color() ?? .red)
```
Comment (Kinark) [inline winston/components/Badge.swift:89]: There's now a component for that glowing dot (that's also used on PostLinks), please replace this ZStack for the component and use .equatable()

File: winston/components/Badge.swift [hunk 2]
```diff
@@ -75,14 +95,11 @@ struct BadgeView: View, Equatable {
...
-              let flairWithoutEmoji = String(f[(f.contains(":") ? f.index(colonIndex!, offsetBy: min(2, max(0, f.count - f.distance(from: f.startIndex, to: colonIndex!)))) : f.startIndex)...])
-              if !flairWithoutEmoji.isEmpty {
-                // TODO Load flair emojis via GET /api/v1/{subreddit}/emojis/{emoji_name}
-                Text(flairWithoutEmoji).font(.system(size: theme.flairText.size, weight: theme.flairText.weight.t)).lineLimit(1).foregroundColor(theme.flairText.color.cs(cs).color()).padding(EdgeInsets(top: 0, leading: 6, bottom: 0, trailing: 6)).background(theme.flairBackground.cs(cs).color()).clipShape(Capsule())
-              }
+          if let flairs = flairWithoutEmojis(str: userFlair) {
+            // TODO Load flair emojis via GET /api/v1/{subreddit}/emojis/{emoji_name}
+            ForEach(flairs, id: \.self) {
+              Text($0).font(.system(size: theme.flairText.size, weight: theme.flairText.weight.t)).lineLimit(1).foregroundColor(theme.flairText.color.cs(cs).color()).padding(EdgeInsets(top: 0, leading: 4, bottom: 0, trailing: 4)).background(theme.flairBackground.cs(cs).color()).cornerRadius(4.0)
+
```
Comment (Kinark) [inline winston/components/Badge.swift:102]: Please make this an Equatable component

**Review by Kinark — CHANGES_REQUESTED** [review]

File: winston/components/Badge.swift (hunk 2 above)
Comment (zanderb98) [reply winston/components/Badge.swift]: Done

File: winston/components/Badge.swift (hunk 1 above)
Comment (zanderb98) [reply winston/components/Badge.swift]: Done!

File: winston/components/Badge.swift [hunk 3]
```diff
@@ -104,12 +146,12 @@ struct BadgeView: View, Equatable {
...
-          //   if unseenComments > 0 {
-          //     Text("(\(Int(unseenComments)))").foregroundColor(.accentColor)
-          //   }
-          // }
+          if postWrapper.hasValue, let seenComments =  postWrapper.post!.data?.winstonSeenCommentCount  {
+            let newComments = Int(commentsCount ?? "0")! - seenComments
+            if newComments > 0 {
+              Text("(\(newComments))").foregroundColor(.accentColor)
+            }
+          }
```
Comment (Kinark) [inline winston/components/Badge.swift:154]: We **cannot** pass the post down to the child views, not even inside wrappers, don't do that, please. Besides, creating a PostWrapper to be passed as an environmentObject every time badge is invoked is a terrible idea. Optional observable objects are bad, we don't try to workaround that.

**Review by Kinark — CHANGES_REQUESTED** [review]

File: winston/components/Badge.swift (hunk 3 above)
Comment (zanderb98) [reply winston/components/Badge.swift]: Sounds good. Reverted those changes for now. We can figure that out in a futurePR

**Review by Kinark — APPROVED** [review]

**Pull request merged**